### DISCUSSION TITLE
2d slices from 3d image shuffle/randomize

### DISCUSSION THREAD

**Discussion by Djennifer** [Project-MONAI/tutorials#1054] [Q&A] [answered]
I'm following the `2d_slices_from_3d_training.ipynb` example and noticed that the slices are in order of the 3D image. In my case I want to minimize that my model will be stuck in a local minima and thus it would be helpful if I was able to shuffle the 2D slices from different images. So that in each batch very different looking 2D slices are present. I just cannot find the function to do something like this. `monai.data.PatchIterd` and `monai.data.GridPatchDataset` don't have a shuffle argument and I can't use the shuffle argument in the Dataloader since I use an IterableDataset

**Comment by wyli** [comment, marked as answer]
in the tutorial https://github.com/Project-MONAI/tutorials/blob/main/modules/2d_slices_from_3d_training.ipynb,

that's done with a `ShuffleBuffer`: `shuffle_ds = monai.data.ShuffleBuffer(patch_ds, buffer_size=30, seed=0)` when the buffer_size is larger than number of slice samples per volume, it'll shuffle across subjects.

**Reply by Djennifer** [reply]
Thank you! I completely missed that line, apparently
